src/main/java/mekanism/common/world/ResizableDiskReplaceFeature.java

```diff
@@ -19,38 +14,21 @@ public ResizableDiskReplaceFeature(Codec<ResizableDiskConfig> codec) {
 
     @Override
     public boolean place(FeaturePlaceContext<ResizableDiskConfig> context) {
-        boolean placed = false;
-        BlockPos pos = context.origin();
-        WorldGenLevel world = context.level();
-        if (world.getFluidState(pos).is(FluidTags.WATER)) {
-            ResizableDiskConfig config = context.config();
-            int halfHeight = config.halfHeight.getAsInt();
-            int yMax = pos.getY() + halfHeight;
-            int yMin = pos.getY() - halfHeight - 1;
-            int radius = config.radius.sample(context.random());
-            int radiusSquared = radius * radius;
-            for (int x = pos.getX() - radius; x <= pos.getX() + radius; ++x) {
-                int xRadius = x - pos.getX();
-                int xRadiusSquared = xRadius * xRadius;
-                for (int z = pos.getZ() - radius; z <= pos.getZ() + radius; ++z) {
-                    int zRadius = z - pos.getZ();
-                    if (xRadiusSquared + zRadius * zRadius <= radiusSquared) {
-                        for (int y = yMax; y > yMin; --y) {
-                            BlockPos targetPos = new BlockPos(x, y, z);
-                            Block blockAtTarget = world.getBlockState(targetPos).getBlock();
-                            for (BlockState target : config.targets) {
-                                if (target.is(blockAtTarget)) {
-                                    world.setBlock(targetPos, config.state, Block.UPDATE_CLIENTS);
-                                    markAboveForPostProcessing(world, targetPos);
-                                    placed = true;
-                                    break;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-        }
-        return placed;
+        ResizableDiskConfig config = context.config();
```

Comment:
If my understanding of escape analysis is correct, these should be stack-allocated if the function is called often enough to be optimized by the JVM.